internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_runtime.go

```diff
@@ -39,7 +39,11 @@ func (r *Runtime) Config() config.Provider {
 func (r *Runtime) SetConfig(b []byte) error {
 	cfg, err := configloader.NewFromBytes(b)
 	if err != nil {
-		return fmt.Errorf("failed to set config: %w", err)
+		return fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	if err := cfg.Validate(r.State().Platform().Mode()); err != nil {
```

Comment:
This is invasive beyond the strict scope of this issue, but IMO, the config should always be validated before it is applied to the Runtime.  As far as I can tell, this is safe everywhere, in that the platform mode is always defined before we set the config.